### DISCUSSION TITLE
duration value seems not accurate

### DISCUSSION THREAD

**Discussion by zfan40** [katspaugh/wavesurfer.js#3656] [Q&A] [answered]
## Bug description
When I use explore waveSurfer and ToneJS, I find the duration not compatible.
<img width="632" alt="image" src="https://github.com/katspaugh/wavesurfer.js/assets/2623884/1df84501-f773-4e2b-aa35-3e6fc606b705">

I am not sure which result is correct, so I dive a little bit and below are three results on the same audio file
[A4.mp3.zip](https://github.com/katspaugh/wavesurfer.js/files/14967944/A4.mp3.zip)

1. ws.getDuration() in events example page
Decode 13.374694s
Ready 13.374694s
<img width="228" alt="image" src="https://github.com/katspaugh/wavesurfer.js/assets/2623884/704366f9-6aa2-43c2-959a-df5c5cb66df3">

2. using python 

```
from pydub import AudioSegment

# Load the audio file
audio_path = "/mnt/data/A4.mp3"
audio = AudioSegment.from_mp3(audio_path)

# Get the duration of the audio file in seconds
duration_seconds = len(audio) / 1000  # Duration in milliseconds to seconds
duration_seconds

```
13.323s

3 using ToneJS
result is 13.32297052154195

It looks like wavesurfer.duration is a bit away from the rest two (but I cannot dictate which is absolutely correct. may need another solid proof)
## Environment
 - Browser: Chrome

## Minimal code snippet
The existing example events.js, just replace the audio file would be fine.

## Expected result
13.323 ish

## Obtained result
13.3746

**Comment by katspaugh** [comment, marked as answer]
You can pass your own duration (e.g. from tonejs) in the options.

**Reply by zfan40** [reply]
thanks, this is actually a solid solution after careful thoughts. it's not wise to keep different source of truth

**Comment by Johnrobmiller** [comment]
This does not work. When I set the duration manually in the options, it still gets the duration wrong. In my case, I manually set the duration in the options as 6.84 but `ws.getDuration()` incorrectly logged 6.81795.

This seems to be a bug to me, created an issue here: https://github.com/katspaugh/wavesurfer.js/issues/3707
